### PR TITLE
Added Help link in Alert Dialogue

### LINKED ISSUE #3433: Help popup in Achievements: Link to tips
In each of the help popups, it would be great to have a link that a wiki page that explains things much more deeply.

For instance the help popup for "Featured Images" could link to https://commons.wikimedia.org/wiki/Commons:Featured_picture_candidates or https://commons.wikimedia.org/wiki/Commons:Featured_pictures

### PR DESCRIPTION
**Pull request by Ayan-10**
**Description**
Added help link to each alert dialogue with custom tab.

Fixes #3433 

In each of the help popups, it would be great to have a link that a wiki page that explains things much more deeply. So I added a link to each info button that shows an alert dialogue and there I add a READ ME button from where the user can open the following custom webpage.

**Tests performed **

Tested 3.0.0-debug-helpLink~afd977f6c on Pixel 3a with API level 27.

**Screenshots (for UI changes only)**

Need help? See https://support.google.com/android/answer/9075928

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 1]
```diff
@@ -510,6 +526,19 @@ private void launchAlert(String title, String message){
                 .create()
                 .show();
     }
+    // Launch Alert with a READ MORE button and clicking it open a custom webpage
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:529]: Use Multiline JavaDoc `/** * */`

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 2]
```diff
@@ -456,44 +465,51 @@ private void hideProgressBar(Achievements achievements) {
...
+            ,THANKS_URL);
     }
 
     @OnClick(R.id.quality_images_info)
     public void showQualityImagesInfo() {
-        launchAlert(getResources().getString(R.string.statistics_quality)
-            , getResources().getString(R.string.quality_images_info));
+        launchAlertWithHelpLink(getResources().getString(R.string.statistics_quality)
+            , getResources().getString(R.string.quality_images_info)
+            ,QUALITY_IMAGE_URL);
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:512]: Space after comma.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 3]
```diff
@@ -456,44 +465,51 @@ private void hideProgressBar(Achievements achievements) {
...
+            , IMAGES_FEATURED_URL);
     }
 
     @OnClick(R.id.thanks_received_info)
     public void showThanksReceivedInfo(){
-        launchAlert(getResources().getString(R.string.statistics_thanks)
-                ,getResources().getString(R.string.thanks_received_explanation));
+        launchAlertWithHelpLink(getResources().getString(R.string.statistics_thanks)
+                ,getResources().getString(R.string.thanks_received_explanation)
+            ,THANKS_URL);
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:505]: Space after comma.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 4]
```diff
@@ -456,44 +465,51 @@ private void hideProgressBar(Achievements achievements) {
...
+            ,IMAGES_USED_URL);
     }
 
     @OnClick(R.id.images_nearby_info)
     public void showImagesViaNearbyInfo(){
-        launchAlert(getResources().getString(R.string.statistics_wikidata_edits)
-                ,getResources().getString(R.string.images_via_nearby_explanation));
+        launchAlertWithHelpLink(getResources().getString(R.string.statistics_wikidata_edits)
+                ,getResources().getString(R.string.images_via_nearby_explanation)
+            ,IMAGES_NEARBY_PLACES_URL);
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:491]: Space after comma.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 5]
```diff
@@ -456,44 +465,51 @@ private void hideProgressBar(Achievements achievements) {
...
+            ,IMAGES_REVERT_URL);
     }
 
     @OnClick(R.id.images_used_by_wiki_info)
     public void showUsedByWikiInfo(){
-        launchAlert(getResources().getString(R.string.images_used_by_wiki)
-                ,getResources().getString(R.string.images_used_explanation));
+        launchAlertWithHelpLink(getResources().getString(R.string.images_used_by_wiki)
+                ,getResources().getString(R.string.images_used_explanation)
+            ,IMAGES_USED_URL);
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:484]: Spaces after comma.
This makes code more readable.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 6]
```diff
@@ -62,6 +61,16 @@
     private static final double BADGE_IMAGE_WIDTH_RATIO = 0.4;
     private static final double BADGE_IMAGE_HEIGHT_RATIO = 0.3;
 
+    // Help link URLs
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:64]: Use official multiline JavaDoc.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java (hunk 1 above)
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:529]: Better to write about the function parameters in Javadoc, helps fellow contributors in understanding its use.

File: app/src/main/res/layout/fragment_achievements.xml [hunk 7]
```diff
@@ -112,20 +112,20 @@
...
               android:progress="50"
-              app:progress_text_size="@dimen/progressbar_text"
               app:progress_end_color="#8C8B98"
               app:progress_start_color="#3A3381"
               app:progress_stroke_width="@dimen/progressbar_stroke"
-              app:progress_text_format_pattern="573/110"
               app:progress_text_color="@color/secondaryColor"
+              app:progress_text_format_pattern="573/110"
+              app:progress_text_size="@dimen/progressbar_text"
               app:style="solid_line" />
```
Comment (4D17Y4) [inline app/src/main/res/layout/fragment_achievements.xml:129]: No major changes in this file, apart from the attributes alignment, if @nicolas-raoul  agrees we can revert them back.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -609,5 +609,6 @@ Upload your first media by tapping on the add button.</string>
   <string name="media_detail_step_title">Media Details</string>
   <string name="menu_view_category_page">View category page</string>
   <string name="menu_view_item_page">View item page</string>
+  <string name="Help_Link_Read">Read More</string>
```
Comment (4D17Y4) [inline app/src/main/res/values/strings.xml:612]: use small letters as we are using underscore.

**Review by 4D17Y4 — COMMENTED** [review]
**Review by student**

**Before merge**
Issue demonstration.

https://user-images.githubusercontent.com/54016427/113446425-671b5280-9415-11eb-831d-6852d4e97164.mp4

**After merge**
Solved issue

https://user-images.githubusercontent.com/54016427/113446427-67b3e900-9415-11eb-88fa-591f10833669.mp4

The pull request solves the issue as expected.
All the links worked perfectly.
Cross-checked the links with @nicolas-raoul 's comment on the issue.

Over to @nicolas-raoul , @neslihanturan .

Thanks.

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java [hunk 9]
```diff
@@ -62,6 +61,16 @@
...
+    private static final String IMAGES_UPLOADED_URL = "https://commons.wikimedia.org/wiki/Commons:Project_scope";
+    private static final String IMAGES_REVERT_URL = "https://commons.wikimedia.org/wiki/Commons:Deletion_policy#Reasons_for_deletion";
+    private static final String IMAGES_USED_URL = "https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Images";
+    private static final String IMAGES_NEARBY_PLACES_URL = "https://www.wikidata.org/wiki/Property:P18";
+    private static final String IMAGES_FEATURED_URL = "https://commons.wikimedia.org/wiki/Commons:Featured_pictures";
+    private static final String QUALITY_IMAGE_URL = "https://commons.wikimedia.org/wiki/Commons:Quality_images";
+    private static final String THANKS_URL = "https://www.mediawiki.org/wiki/Extension:Thanks";
+
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:73]: Just one new line is enough :-)

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java (hunk 5 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:484]: @4D17Y4 is right that space is needed after every comma.
Actually the best here would be to put the commas at the end of each line.
Also, indentation should be the same for arguments of the same level.
Not a hard convention but when there are arguments over several lines and most arguments are rather long, for readibility I suggest using one line per argument.
In other words, I recommend changing to the following, thanks!
Screen Shot 2021-04-03 at 11 24 27

File: app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java (hunk 1 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:529]: +1 to what @4D17Y4 wrote, plus one new line to separate from the previous method.

File: app/src/main/res/layout/fragment_achievements.xml (hunk 7 above)
Comment (nicolas-raoul) [reply app/src/main/res/layout/fragment_achievements.xml:129]: I agree with @4D17Y4 , if there are no "real" change (meaning the resulting APK is strictly equivalent), then better do such beautification in a separate pull request.

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (nicolas-raoul) [reply app/src/main/res/values/strings.xml:612]: Help_Link_Read -> read_help_link
Read More -> Read more

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Thanks @4D17Y4 for your great pull request review!
Thanks @Ayan-10 for the pull request, no major problem but would you mind fixing the minor noted issues?

**Comment by Ayan-10** [comment]
Thanks @4D17Y4 @nicolas-raoul , For this great review. I will keep these things in mind for my next PRs. I hope I fixed all minor issues as listed. Sorry for the inconvenience.

**Review by nicolas-raoul — APPROVED** [review]
Looks great to me. Thanks!

**Pull request merged**